lib/usd/translators/meshWriter.cpp

```diff
@@ -163,13 +284,171 @@ bool PxrUsdTranslators_MeshWriter::writeMeshAttrs(
+
+                    if (curIntermediate.hasFn(MFn::kTweak)) {
+                        // NOTE: (yliangsiew) Make sure tweak really has no effect even if it's on
+                        MPlug plgVLists = fnGeoFilt.findPlug("vlist");
+                        assert(plgVLists.isArray());
+                        unsigned int numPlgVlists = plgVLists.numElements();
+                        for (unsigned int j = 0; j < numPlgVlists; ++j) {
+                            MPlug plgVlist = plgVLists.elementByPhysicalIndex(j); // vlist[0]
+                            assert(plgVlist.isCompound());
+                            unsigned int plgVlistNumChildren = plgVlist.numChildren();
+                            for (unsigned int k = 0; k < plgVlistNumChildren; ++k) {
+                                MPlug plgVlistChild = plgVlist.child(k); // vlist[0].vertex
+                                assert(plgVlistChild.isArray());
+                                unsigned int numPlgVlistChildElements = plgVlistChild.numElements();
+                                for (unsigned int x = 0; x < numPlgVlistChildElements; ++x) {
+                                    MPlug plgVertex = plgVlistChild.elementByPhysicalIndex(
+                                        x); // vlist[0].vertex[0]
+                                    assert(plgVertex.isCompound());
+                                    unsigned int plgVertexNumChildren = plgVertex.numChildren();
+                                    for (unsigned int y = 0; y < plgVertexNumChildren; ++y) {
+                                        MPlug plgVertexComponent
+                                            = plgVertex.child(y); // vlist[0].vertex[0].xVertex
+                                        float vertexComponent = plgVertexComponent.asFloat();
+                                        if (fabs(vertexComponent - 0.0f) > FLT_EPSILON) {
+                                            MFnDependencyNode fnNode(curIntermediate, &status);
+                                            CHECK_MSTATUS_AND_RETURN(status, false);
+                                            MGlobal::displayError(
+                                                "Could not determine the original blendshape "
+                                                "source mesh due to the tweak node: "
+                                                + fnNode.name()
+                                                + ". Please either bake it down or remove the "
+                                                  "edits and attempt the export process again, or "
+                                                  "specify -ignoreWarnings.");
+                                            TF_RUNTIME_ERROR(
+                                                "Could not determine the original blendshape "
+                                                "source mesh due to a non-empty tweak node, "
+                                                "aborting export.");
```

Comment:
Are both of these necessary? I would have thought that the `TF_RUNTIME_ERROR` would be sufficient, and that the diagnostic delegate we install would take care of surfacing the message using `MGlobal::displayError()`:
https://github.com/Autodesk/maya-usd/blob/1439297b4ba79bdcffcfa6551b143ea7ac46f99a/lib/mayaUsd/utils/diagnosticDelegate.cpp#L104

And then a few more cases of this below.

Reply:
I actually did not see any status messages show up in Maya when using TF_RUNTIME_ERROR (they only show up in the terminal stdout), which is why I was making explicit calls to display the messages as well. If there's something I'm missing here like a config or a compile flag, I'd love to know so I don't have to essentially duplicate these calls.

Reply:
Ahh, sorry. Apparently we currently only install the diagnostic delegate I was talking about for the `pxrUsd` plugin:
https://github.com/Autodesk/maya-usd/blob/dev/plugin/pxr/maya/plugin/pxrUsd/plugin.cpp#L164

The `mayaUsdPlugin` does **not** install that delegate, which is probably why the `TF_RUNTIME_ERROR` doesn't get translated into a Maya error message.

Reply:
Just posted #1003 which I think should address this and let us remove the extra `MGlobal::...` calls.